File: python/registro.py
'''
Registro de alumnos
'''
from utilidades import linea, encabezado, Any

def main( ) -> None:
  #definicion de variables
  ci: str = '34.860.848'
  name: str = 'Alfonso'
  last_name: str = 'Henriquez'
  age: int = 17
  height: float = 1.69
  gender: bool = False #True = femenino
  turn: int = 1 #1 = mañana 2 = tarde y 3 = noche
  view_info(ci, name, last_name, age, height, gender, turn)
  
def view_info(*datos: Any) -> None: #se le indica con *datos = a todos los datos q se le agreguen los convierta en una tupla
  encabezado('Datos de alumnos', 50)
  print (f'Cédula: {datos[0]}')
  print (f'Nombre completo {datos[1].upper()} {datos[2].upper()}')
  print (f'Tiene {datos[3]} años')
  print (f'Su altura es de {datos[4]}m')
  print (f'Genero: {gender(datos[5])}')
  print (f'{turno(datos[6])}')
  linea(50)
  
def gender(gender: bool) -> str:
  if gender:
    return 'femenino'
  else: 
    return 'masculino'
  
def turno(turno: int) -> str:
  match turno:
    case 1:
      return "Turno de la mañana"
    case 2:
      return "Turno de la tarde"
    case 3:
      return "Turno de la noche"
    case _: #piso si no tiene valor
      return 'No tiene turno'
  
main() 

  

File: Js/src/registro.js
/* 
Programa para registrar alumnos en JS
*/

import { encabezado, line, capitalizar } from "./utilidades.js";

const main = () => {

  const view_info = (...datos) => {
    encabezado('Registro del alumno', 32)
    console.log(`Cédula: ${datos[0]}`)
    console.log(`Nombre completo: ${capitalizar(`${datos[1]}`)} ${capitalizar(`${datos[2]}`)}`)
    console.log(`Tiene ${datos[3]} años`)
    console.log(`Su altura es de ${datos[4]}m`)
    console.log(`Genero: ${genero(datos[5])}`)
    console.log(`${turno(datos[6])}`)
  }

  const genero = (gender) => {
    if (gender) {
      return 'femenino'
    }
    else {
      return 'masculino'
    }
  }

  const turno = (turn) => {
    switch (turn) {
      case 1:
        return "Turno de la mañana"
      case 2:
        return "Turno de la tarde"
      case 3:
        return "Turno de la noche"
      default:
        return "no tiene turno"
    }
  }

  // variables de alumno
  let ci = '34.860.848';
  let name = 'alfonso';
  let last_name = 'henriquez';
  let age = 18;
  let gender = true;
  let height = 1.69;
  let turn = 1;  //1 = mañana 2 = tarde y 3 = noche
  view_info(ci, name, last_name, age, height, gender, turn);


}

main()